src/OpenTelemetry.Contrib.Instrumentation.Quartz/Implementation/QuartzDiagnosticListener.cs

```diff
@@ -0,0 +1,102 @@
+// <copyright file="QuartzDiagnosticListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using OpenTelemetry.Instrumentation;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Quartz.Implementation
+{
+    internal sealed class QuartzDiagnosticListener : ListenerHandler
+    {
+        internal static readonly AssemblyName AssemblyName = typeof(QuartzDiagnosticListener).Assembly.GetName();
+        internal static readonly string ActivitySourceName = AssemblyName.Name;
+        internal static readonly Version Version = AssemblyName.Version;
+        internal static readonly ActivitySource ActivitySource = new ActivitySource(ActivitySourceName, Version.ToString());
+
+        private readonly QuartzInstrumentationOptions options;
+
+        public QuartzDiagnosticListener(string sourceName, QuartzInstrumentationOptions options)
+            : base(sourceName)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            if (activity.IsAllDataRequested)
+            {
+                if (this.options.TracedOperations != null && !this.options.TracedOperations.Contains(activity.OperationName))
+                {
+                    QuartzInstrumentationEventSource.Log.RequestIsFilteredOut(activity.OperationName);
+                    activity.IsAllDataRequested = false;
+                    return;
+                }
+
+                activity.DisplayName = this.GetDisplayName(activity);
+
+                ActivityInstrumentationHelper.SetActivitySourceProperty(activity, ActivitySource);
+                ActivityInstrumentationHelper.SetKindProperty(activity, this.GetActivityKind(activity));
+            }
+        }
+
+        public override void OnException(Activity activity, object payload)
+        {
+            if (!this.options.TracedOperations.Contains(activity.OperationName))
+            {
+                return;
+            }
+
+            if (!(payload is Exception exception))
```

Comment:
this seems different that the fetcher approach used in other instrumentations.. https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L271

Reply:
i'd recommend a unit test to see if this works.

Reply:
@cijothomas , I have been looking at the example you have linked. 

`  private readonly PropertyFetcher<Exception> stopExceptionFetcher = new PropertyFetcher<Exception>("Exception");`

When processing the `stopExceptionFetcher.TryFetch`
`  if (!this.stopExceptionFetcher.TryFetch(payload, out Exception exception) || exception == null) `

The fetch returns null even though the `payload` is of type `Exception`. I think that this is because the `stopExceptionFetcher` is looking for a property `Exception` on the exception object.